k8s/upgrades/jiva-0.6.0-0.8.1/README.md

```diff
@@ -0,0 +1,145 @@
+
+Before upgrading the OpenEBS Operator, check if you are using a storage pool named `default` which will conflict with default jiva pool installed with OpenEBS 0.8.1:
+```
+./pre_upgrade.sh <openebs-namespace>
+```
+
+#### Storage Classes
+OpenEBS supports specified Storage Policies in Storage Classes. The way storage policies are specified has changed in 0.7.x. The policies will have to be specified under metadata instead of parameters. 
+
+For example, if your storage class looks like this in 0.6.0:
+```
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+   name: openebs-mongodb
+provisioner: openebs.io/provisioner-iscsi
+parameters:
+  openebs.io/storage-pool: "default"
+  openebs.io/jiva-replica-count: "3"
+  openebs.io/volume-monitor: "true"
+  openebs.io/capacity: 5G
+  openebs.io/fstype: "xfs"
+```
+
+There is no need to mention the volume-monitor and capacity with 0.7.0. The remaining policies like storage pool, replica count and the fstype should be specified as follows:
+```
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+   name: openebs-mongodb
+   annotations:
+    cas.openebs.io/config: |
+      - name: ReplicaCount
+        value: "3"
+      - name: StoragePool
+        value: default
+      - name: FSType
+        value: "xfs"
+provisioner: openebs.io/provisioner-iscsi
+```
+
+Make edits to your Storage Class YAMLs - delete them and add them back. A delete and re-apply is required since updates to Storage Class parameters are not possible. 
+
+If you are using `ext4` for FSType, you could use the following script to upgrade your StorageClasses. 
+```
+./upgrade_sc.sh
+```
+
+Alternatively, you can skip this step and re-apply your StorageClasses as per the 0.7.0 volume policy specification. 
+
+**Important Note: StorageClasses have to updated prior to provisioning any new volumes with 0.7.0.**
+
+## Step 1: Upgrade the OpenEBS Operator
+
+### Upgrading OpenEBS Operator CRDs and Deployments
+
+The upgrade steps vary depending on the way OpenEBS was installed, select one of the following:
+
+#### Install/Upgrade using kubectl (using openebs-operator.yaml )
+
+**The sample steps below will work if you have installed openebs without modifying the default values in openebs-operator.yaml. If you have customized it for your cluster, you will have to download the 0.7.0 openebs-operator.yaml and customize it again**
+
+```
+#Upgrade to 0.8.1 OpenEBS Operator
+kubectl apply -f https://openebs.github.io/charts/openebs-operator-0.8.1.yaml
+```
+
+#### Install/Upgrade using helm chart (using stable/openebs, openebs-charts repo, etc.,) 
+
+**The sample steps below will work if you have installed openebs with default values provided by stable/openebs helm chart.**
+
+- Run `helm repo update` to update local cache with latest package
+- Run `helm ls` to get the release name of openebs. 
```

Comment:
do we need to change any helm upgrade steps here @prateekpandey14 ?

Reply:
Update from 0.5.0 had values suggested during update process (see https://github.com/openebs/openebs/tree/master/k8s/upgrades/0.5.x-0.6.0#installupgrade-using-helm-chart-using-stableopenebs-openebs-charts-repo-etc and https://openebs.github.io/charts/helm-values-0.6.0.yaml). If anyone updated previously using those steps, he'll have all values outdated. 

I'd at least mention that, or even add `--reset-values` into the installation for those who applied values above. Otherwise upgrade will pass, but many image tags will stay at `0.6.0`.

Reply:
Thanks @rgembalik for reviewing! How about adding the following note. 

---
Before upgrading using helm, please review the default values available with latest stable/openebs chart. (https://raw.githubusercontent.com/helm/charts/master/stable/openebs/values.yaml). 
- If the default values seem appropriate, you can use the `helm upgrade --reset <release name> stable/openebs`. 
- If not, customize the values into your copy (say `custom-values.yaml`), by copying the content from above default yamls and edit the values to suite your environment. You can upgrade using your custom values using:
  `helm upgrade <release name> stable/openebs -f custom-values.yaml`

Reply:
Looks good to me, although I am 99% sure that it should say `helm upgrade --reset-values <release name> stable/openebs` instead of `helm upgrade --reset <release name> stable/openebs`

Reply:
yes! you are right.